### DISCUSSION TITLE
Karşılık seçme ilkeleri

### DISCUSSION THREAD

**Discussion by Cuneytakin** [turkce-yazilim-konati/yazilimca#67] [Türkçe]
Yazılım dilindeki yabancı sözcüklere karşılık seçme aşamasında, yerleşmiş karşılıkları değiştirmek doğru olmayabilir. Mesela, base, taban demek, link bağlantı demek, bunlar yerine yeni sözcük üretmek bana anlamlı ve işlevsel gelmiyor doğrusu.

---
**_Yazılımcanın bilegeninde karşılık seçme ilkeleri ayrıntılarıyla yazılacaktır. Erişmek için bağlantı örkü:_** [Bilegen: Karşılık Seçme İlkeleri](https://github.com/turkce-yazilim-konati/yazilimca/wiki/Kar%C5%9F%C4%B1l%C4%B1k-Se%C3%A7me-%C4%B0lkeleri)

**Comment by volkantash** [comment]
Karşılık seçme ilkeleri konusunda Türk Dil Kurumu uzmanı Sn. Belgin Tezcan Aksu'nun şu sunumunda değindiklerinden de yararlanabiliriz.
https://www.youtube.com/watch?v=0QiuqwPv25A

**Reply by turkmenistanli** [reply]
Bu konuda bir sistemin oturtulması şart. Eğer bu sözcüklere karşılık bulunacaksa da karşılıkların dikkatli seçilmesi gerekecek.

**Comment by volkantash** [comment]
https://github.com/turkce-yazilim-konati/yazilimca/issues/10#issuecomment-1263861568
> Bence bu kelimeleri seçerken en kısası işimizi görüyorsa onu seçmeliyiz yani ekle demek yerine ek demek uygunsa bunu kullanmalıyız. Kod yazan kişi için çok iyi olur. Bahaettiğim şekilde kısa kelimeler seçilirse kod yazım aşamasında pratik olur. Bu konuda seçim yaparken bu hususa dikkat ediyormusunuz?



Kelime seçmede kısalık uzunluk seçmeyi etkileyen bir ölçü sayılmamalı. Doğru bir ad verilmelidir. Kısalık uzunluk örneği olmasa da karşılık arayışlarında printer sözcüğüne yazıcı sözcüğü ile karşılık verilmesi örneğini sık sık dile getiririz. Bu "birebir karşılık vermeyelim" gibi yanlışlar doğuran bir anlayışa sarılmanın sonucudur. Şimdi "3D Printer" için Türkçe olarak "yazıcı" karşılığını kullanamıyorlar. Bunun gibi birçok örnek vardır. Bence kısalık uzunluk ölçülerine göre adlar, terimler seçer isek bu ne bilimsel bir yaklaşım olur, ne çözücü bir yaklaşım olur, ne de bilgece bir yaklaşım olur. 

Programlayıcının eli yorulmasın diye endişe duyup beslemek yerine düşünce dili Türkçe olan programlayıcının düşüncelerini sıkmayan, darlamayan, akağan kılan bir düzende olup olmamayı önde tutmalıyız.

Evet, sözdizimi gibi konular başta olmak üzere ad belirleyip koymak bu alanın en çok çaba isteyen alanıdır. Ne diyeceğimizi bildikten sonra işler kolay kalır. Ancak ne diyeceğimizi bilene dek geçen sürede işler kolay kalmayacaktır.

Adları seçme konusunda özet ile demek gerekir ise kısadır, uzundur gibi bir yaklaşımı bu alanda kendime bir ölçü olarak koymadım. Benim gözümde bağlayıcı olanlar başka ölçüler. Ön verirsiniz ki herkesin bu söylediklerim ile oydaş olduğunu dillendiremem.

**Comment by osmankalhan** [comment]
Mesela örnek verdiğiniz Link : Baglanti kısmı çok güzel bir türkçeleştirme. Ama bahsettiğiniz base : taban eşleştirmesi türkçeleştirilmesi kafa karışıklığına yol açabilecek bir türkçeleştirme olabilir. Bu tür karşılık seçmelerde yüzlerce kişiden fikirler alarak bunları ön elemeden geçirip mantıklı olan 10 taneyi oylamaya sunma yöntemi ile seçip uygulayabiliriz.
